lib_eio/task_pool.mli

```diff
@@ -0,0 +1,9 @@
+type 'a task = unit -> 'a
+
+type t
+
+val create : max_domains:int -> Domain_manager.t -> t
```

Comment:
There needs to be a switch argument here. Otherwise, any worker domain created can't outlive the `create` or `run` call, which defeats the point of a pool.